src/libostree/ostree-repo.c

```diff
@@ -933,6 +933,44 @@ ostree_repo_write_config (OstreeRepo *self,
 {
   g_return_val_if_fail (self->inited, FALSE);
 
+  /* Ensure that any remotes in the new config aren't defined in a
+   * separate config file.
+   */
+  g_auto(GStrv) groups = NULL;
+  gsize num_groups, i;
+  groups = g_key_file_get_groups (new_config, &num_groups);
+  for (i = 0; i < num_groups; i++)
+    {
+      g_autoptr(OstreeRemote) new_remote = NULL;
+      g_autoptr(OstreeRemote) cur_remote = NULL;
+
+      new_remote = ostree_remote_new_from_keyfile (new_config, groups[i]);
+      if (new_remote != NULL)
+        {
+          g_autoptr(GError) local_error = NULL;
+
+          cur_remote = _ostree_repo_get_remote (self, new_remote->name,
+                                                &local_error);
+          if (cur_remote == NULL)
+            {
+              if (!g_error_matches (local_error, G_IO_ERROR,
+                                    G_IO_ERROR_NOT_FOUND))
+                {
+                  g_propagate_error (error, g_steal_pointer (&local_error));
+                  return FALSE;
+                }
+            }
+          else if (cur_remote->file != NULL)
+            {
+              g_set_error (error, G_IO_ERROR, G_IO_ERROR_EXISTS,
```

Comment:
I'm not sure `G_IO_ERROR_EXISTS` is the error that should be thrown here. Suggestions welcome.